pkg/v1/providers/tests/clustergen/run_tests.sh

```diff
@@ -77,6 +77,8 @@ generate_cluster_configurations() {
          cat "$t" | perl -pe 's/--plan (\S+)/--plan $1cc/; s/_PLAN: (\S+)/_PLAN: $1cc/' > /tmp/test_tkg_config_cc
          echo "CLUSTER_CLASS: tkg-${infra}-default" >> /tmp/test_tkg_config_cc
          read -r -a cmdargs < <(grep EXE: /tmp/test_tkg_config_cc | cut -d: -f2-)
+         TKR_DATA="  v1.21.2:\n    kubernetesSpec:\n      version: v1.21.2\n      imageRepository: projects-stg.registry.vmware.com\n      etcd:\n        imageTag: v1.0.0-test\n      coredns:\n        imageTag: v1.1.0-test\n      kube-vip:\n        imageTag: v2.0.0-test\n    labels:\n      os-name: ubuntu\n      os-type: linux\n      os-arch: amd64\n  v1.23.5+vmware.1:\n    kubernetesSpec:\n      version: v1.23.5+vmware.1\n      imageRepository: projects-stg.registry.vmware.com\n      etcd:\n        imageTag: v1.0.0-test\n      coredns:\n        imageTag: v1.1.0-test\n      kube-vip:\n        imageTag: v2.0.0-test\n    labels:\n      os-name: ubuntu\n      os-type: linux\n      os-arch: amd64\n    osImageRef:\n      id: test-ami-id\n      region: test-region\n      sku: test-sku\n      publisher: test-publisher\n      offer: test-offer\n      version: test-version\n      thirdPartyImage: test-third-party-image"
```

Comment:
this is pretty hard to read. can we use heredoc instead?
Also, does this need to be done differently for different infras?

Reply:
Added a heredoc. The data can be the same because it is a frankenstein monster of all the data for all the providers. Each provider pulls out the data that they require.